core/types/receipt.go

```diff
@@ -526,9 +526,11 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			feeScalar := new(big.Float).Quo(fscalar, fdivisor)
 			for i := 0; i < len(rs); i++ {
 				if !txs[i].IsDepositTx() {
+					time := uint64(txs[i].time.Unix())
```

Comment:
This `time` is an internal non-consensus field used for spam detection. It is not going to be accurate here.

Reply:
I think you need to modify the `DeriveFields` function to take the timestamp. It's kinda annoying that we now use a time based fork in a place that doesn't keep references to time.

Reply:
Ah nice catch, didn't realize that. Fixed.